### DISCUSSION TITLE
How to deal with Screen on/off

### DISCUSSION THREAD

**Discussion by GekoCH** [TheTimeWalker/wallpanel-android#35] [Q&A]
Hy

I'm a bit confused how to setup my android to get following function:
I like to wake up the screen via MQTT (this currently works great)
I like to witch off the screen via MQTT (how do I need to set the system settings and the app settings so that I can send the {"wake": false} command to switch the screen off ??)

Thx for any suggestions
Andy

**Comment by Clement-B** [comment]
Unfortunalty, it seems not possible right now with the app or I didnt find the way and would love to know.

If it can help, for achieving this, I use another application called Tasker where you send Intent action from other application (for me Home Assistant), Tasker app then catch the intent and do whatever you want, like switching it off.

**Comment by GekoCH** [comment]
> Unfortunalty, it seems not possible right now with the app or I didnt find the way and would love to know.
> 
> If it can help, for achieving this, I use another application called Tasker where you send Intent action from other application (for me Home Assistant), Tasker app then catch the intent and do whatever you want, like switching it off.

Thx, do you send the command from HA over mqtt or how can tasker listen to a command comming from HA?

**Comment by Clement-B** [comment]
I send it from HA.

You can send broadcast intents as describing here: https://companion.home-assistant.io/docs/notifications/notification-commands/#broadcast-intent

I forgot to mention that you need Home Assistant Companion (android app) installed on the device.

Here is an example of an intent send to my tablet from HA using a script : 

![image](https://user-images.githubusercontent.com/5446081/192163773-1c1e240a-853e-41e6-a5d8-898532bc0aec.png)

Then, in Tasker you need to create a task which would receive this intent and switch off the screen of the device (didnt take time to make screenshot here sorry).
Documentation of Tasker intent : https://tasker.joaoapps.com/userguide/en/intents.html

To come back to the real issue, that Wall Panel can't switch off tablet over HTTP or MQTT :  
* If I get it correctly, in his current state, wall panel dont screen off the device but instead define the sleep time before device automatically switch off the screen of the device.
* To my understanding, it can be challenging to force screen off of device depending of the Android version and it can require specifics permissions. Also, there is many applications that already support this kind of feature.
* For a basic usage of the app, switch off the screen is normally not needed as default sleep time of device would eventually switch it off.

So maybe it's more of a feature request here ?

For me, I had some use cases where I wanted to specifically switch off the screen of my wall panel device, which were : 
* Sometimes, maybe caused by camera stuck on detection, my device would not normally screen off ;
* My wall tablet is near my TV, so when I start to watch something, I wanted to screen off the device in addition to disable camera detection ;
* When I enable / disable charging of the device, tablet would automattically screen on, so I created a routine to automatically screen off the device after the action.

Hope it can help you and I would love to know how others people have achieved this differently, maybe there is an even easy way !

**Comment by GekoCH** [comment]
wow thx a lot for this detailed explanation.

In the mean time I was able to fix it myself.
I have MQTT client and Tasker running on my wallpanel. I even rooted my tablet so I can disable the lock screen. So it is even easier that way since I don't have to "swipe away" the lock screen after waking up the tablet.
Now I can send MQTT commands to the wallpanle that are then executed by tasker like screen on / screen off etc.

**Comment by hawkeye217** [comment]
Maybe one of the devs can answer this question...

I'm using an external PIR sensor and some logic in Home Assistant to keep track of room presence and I'd like to turn on and off my Fire tablet's display when I enter and leave the room. 

I set my tablet to turn off the screen after 15 seconds of inactivity, and I've noticed that if I repeatedly send a ``{ "wake" : true, "waketime" : 180 }`` to the mqtt ``command`` topic (with a 30-60 second delay in between each), I can keep the display on - most of the time.

Every once and a while, the display turns off (seemingly randomly) and doesn't wake up until the next time the mqtt message is sent. All motion and audio detection settings are disabled in the app's settings. 

How does the wake timer work? I would assume that a 180 second wake time being sent every 30 seconds would keep the screen on until I stopped sending it. 

Is there a better way to reliably keep the screen on and turn it off via mqtt?

**Reply by imaginator** [reply]
I'd also love to know this. Currently I run the web screensaver to keep the displays on which keeps wifi on. But it would be much nicer to have a more elegant solution of displays sleeping while wifi stays on.

**Reply by EmJay276** [reply]
@imaginator 
If you have a Fire tablet, there is an setting to keep wifi active while the display is off.
`Settings` => `Power` or `Battery` => `Automatic Smart Suspend` => Turn it off
https://www.amazon.com/-/de/gp/help/customer/display.html?nodeId=GBZFGATCC6JUTWJX

**Comment by jtaseff** [comment]
I'm also stuck on this.... running android 6.0 on a very old tablet, not sure if that makes a difference.
It seems the app itself does have control over what I want - the toggle for "keep screen on" in the settings will permanently keep it on. But then the "wake" command via MQTT doesn't control the same thing.
If the "keep screen on" is on, sending "wake:false" will never let it sleep. If "keep screen on" is off, sending "wake:true" will wake it up, but only for a few minutes until android turns it off again.
Is there a way to control this toggle via command instead?

**Reply by pravardhanreddy** [reply]
Same Issue. Would love to have this

**Comment by EmJay276** [comment]
Toggle the "keep screen on" setting via MQTT would add a real benefit for automation via HomeAssistant!
For example via an additonal `{"keepScreenOn": true/false}` command

I could turn off the setting if I am leaving the house (e.g. `{"keepScreenOn": false}`) => Display will turn off (completly) after Android timeout
When I come back, I send e.g. `{"keepScreenOn": true, "wake": true}` to turn the setting back on and keep the display/screensave active while I am at home.

**Comment by fokusas-IoT** [comment]
Currently this problem with not extending wake time still exists. I have done workaround on my setup with appdaemon (or you can use any other automation). When movement detected i call wake with 36000 seconds. Then I have run_in timer. This timer is recreated each time when movement is detected. Timer sends wake:false when it kicks in.

this is snippet of my code:
```
self.call_service("mqtt/publish", topic="wallpanel/tablet/command", payload=self.to_json({"wake": True, "wakeTime": 36000}))
if self.motion_timer is not None:
    self.cancel_timer(self.motion_timer)
self.motion_timer = self.run_in(lambda kwargs: self.call_service("mqtt/publish", topic="wallpanel/tablet/command", payload=self.to_json({"wake": False})), 300)
```
